.github/workflows/notebook-test.yml

```diff
@@ -68,9 +68,25 @@ jobs:
           args = ["tox", "--", "--write"]
```

Comment:
This has gotten complex enough that I think it could also benefit from being a script under `scripts/ci`.

Btw, you can have the changed files action write its output to a file. Imo it's more maintainable than this string interpolation. See https://github.com/Qiskit/documentation/pull/2389